.github/workflows/artifact.yml

```diff
@@ -11,30 +11,40 @@ jobs:
       packages: write
     steps:
       - name: Checkout
-        uses: actions/checkout@v4.1.0
+        uses: actions/checkout@v4.1.1
         with:
           fetch-depth: 0 # fetch-depth 0 needed for NBGV
       - name: Use Node.js 20
-        uses: actions/setup-node@v3
+        uses: actions/setup-node@v4.0.2
         with:
           node-version: 20
           registry-url: 'https://npm.pkg.github.com'
       - name: Nerdbank.GitVersioning
         id: nbgv
-        uses: dotnet/nbgv@v0.4.1
+        uses: dotnet/nbgv@v0.4.2
         with:
-          stamp: package.json
-      - run: npm ci
-      - run: npm run build
-      - run: npm publish
+          setAllVars: true
+      - name: Install Dependencies
+        run: npm ci
+      - name: Stamp Packages Version
+        run: npm version --workspaces --include-workspace-root --no-git-tag-version "$NBGV_NpmPackageVersion"
+        env:
+          NODE_AUTH_TOKEN: ${{ secrets.GITHUB_TOKEN }}
+      - name: Build Packages
+        run: npm run build --workspaces --include-workspace-root
+      - name: Pack Packages
+        run: |
+          mkdir ./out
+          npm pack --workspaces --include-workspace-root --pack-destination ./out
+      - name: Upload Packages artifact
+        uses: actions/upload-artifact@v4.3.1
+        with:
+          name: packages
+          path: out/dbos-inc-*.tgz
+      - name: Publish Packages
+        run: npm publish --workspaces --include-workspace-root
```

Comment:
Do we still need to publish to Github private packages? I thought we decided to move to NPM for everything..

Reply:
I can remove that, though it doesn't cost us anything to use the GitHub package manager. I think it's a reasonable idea for us to have a feed with all of our pacakges published. Esp after we get to 1.0, we'll need to be more careful about how we publish